torch/csrc/autograd/variable.cpp

```diff
@@ -175,8 +187,12 @@ void Variable::Impl::set_data(const at::Tensor &new_data) {
   type_id_ = new_data.type().type_id();
   is_variable_ = true;
 
+  // `var.set_data(...)` is not supposed to affect `var`'s version counter, so we save
+  // the original version counter and restore it after replacing the underlying tensor.
+  auto saved_version_ = data_.current_version();
   auto new_data_copy = at::Tensor(new_data.getIntrusivePtr()->shallow_copy_and_detach());
   data_ = std::move(new_data_copy);
+  data_.set_version_counter(saved_version_);
```

Comment:
I'm confused by this. I don't see any mutating operations, so I wouldn't expect that the version counter wouldn't get updated in this case. Are you trying to prevent sharing from taking place? But surely `set_version_counter` isn't the appropriate way to go about doing this?

Reply:
Fixed